app/main/user-agent-service.js

```diff
@@ -66,23 +66,30 @@ function configure(app: any, storage: ProfileStorage) {
     return { stars, recentStars };
   }
 
-  const diffsClients = [];
-  router.ws('/diffs', async function(ws, _req) {
-    diffsClients.push(ws);
+  const wsClients = [];
+  router.ws('/ws', async function(ws, _req) {
+    wsClients.push(ws);
 
-    ws.send(JSON.stringify({ type: 'initial', payload: await initial() }));
+    ws.send('protocol', 'v1', wsClients.length);
+
+    ws.send('diff', JSON.stringify({ type: 'initial', payload: await initial() }));
 
     ws.on('close', () => {
-      const index = diffsClients.indexOf(ws);
+      const index = wsClients.indexOf(ws);
       if (index > -1) {
-        diffsClients.splice(index, 1);
+        wsClients.splice(index, 1);
+      }
+
+      if (wsClients.length === 0) {
```

Comment:
I'm pretty sure we want the service to outlive clients and be tied to the both client counts and the launching process, but I'm fine with this for now.